landoscript/src/landoscript/actions/tag.py

```diff
@@ -1,17 +1,58 @@
+from dataclasses import dataclass
+
+from aiohttp import ClientSession
 from scriptworker.client import TaskVerificationError
+from scriptworker.utils import retry_async
 
+from landoscript.errors import LandoscriptError
 from landoscript.lando import LandoAction
 
 
-def run(tags: list[str], target_revision: str | None = None) -> list[LandoAction]:
-    if len(tags) < 1:
+# We have two tag info classes to make it easy to signify whether the provided
+# revision is an hg one or a git one. At the time of writing, we use the former
+# for direct tagging tasks (such as `release-early-tagging`), and the latter
+# for merge day operations, where the task payload does not contain a revision.
+# When we switch all consumers (namely, Gecko) to firing their decision tasks
+# off of Github events, we can drop `HgTagInfo`.
+@dataclass(frozen=True)
+class HgTagInfo:
+    revision: str
+    hg_repo_url: str
+    tags: list[str]
+
+
+@dataclass(frozen=True)
+class GitTagInfo:
+    revision: str
+    tags: list[str]
+
+
+async def run(session: ClientSession, tag_info: HgTagInfo | GitTagInfo) -> list[LandoAction]:
+    if len(tag_info.tags) < 1:
         raise TaskVerificationError("must provide at least one tag!")
 
+    git_commit = None
+    if isinstance(tag_info, GitTagInfo):
+        git_commit = tag_info.revision
+    else:
+        # tag_info.revision is an hg revision; lando wants a git revision
+        resp = await retry_async(
+            session.get,
+            args=(f"{tag_info.hg_repo_url}/json-rev/{tag_info.revision}",),
+            kwargs={"raise_for_status": True},
+        )
+        hg_revision_info = await resp.json()
+        # TODO: fail if `git_commit` is None. We can only start doing this
```

Comment:
I'll call out this TODO: explicitly. I expect to address it on Wednesday, after Try has been cut over; possibly before if I don't think I'll be testing this action again before cut-over.

Reply:
What happens right now if git_commit is None?  Looks like we add a tag action with no target?

Reply:
Yeah, which will tag the tip of the branch.

I suppose I can address this now, it will just need reverting to test on Try in the near term...